lib/server.js

```diff
@@ -496,6 +498,14 @@ app.get('/app/cartoref.js', function(req, res, next) {
     res.send('window.cartoRef = ' + JSON.stringify(require('carto').tree.Reference.data) + ';');
 });
 
+app.get('/app/sourceref.js', function(req, res, next) {
+    res.set({
+        'content-type':'application/javascript',
+        'cache-control':'max-age=3600'
+    });
+    res.send('window.sourceRef = ' + JSON.stringify(require('mapnik-reference').load('3.0.0').datasources) + ';');
```

Comment:
Awesome this is moving. One recommendation: Lift the `mapnikReference.load` into a top-level variable so that it is only called once when Studio opens. Also it should be `mapnikReference.load(mapnik.versions.mapnik)` so that the `3.0.0` is not hardcoded and so that the reference dynamically changes based on the Mapnik version studio is running.